### ISSUE TITLE
Server crashed after working on a reactor

### ISSUE THREAD

**Issue by TheRealDadbeard** [BiggerSeries/BiggerReactors#35]
Server crashed kinda two times. First something happened and then it took the whole server down. This are the crash logs from the server. Client only got disconnected.

[crash-2022-01-07_02.44.17-server.txt](https://github.com/BiggerSeries/BiggerReactors/files/7825773/crash-2022-01-07_02.44.17-server.txt)
[crash-2022-01-07_02.45.19-server.txt](https://github.com/BiggerSeries/BiggerReactors/files/7825774/crash-2022-01-07_02.45.19-server.txt)

**Comment by hexxod** [comment]
I had the same issue just now. Don't have the logs though since this was on a friends server and he's not up yet, but it was a hard crash rather than client side.

I'll try and get the logs from him, but the reactor I was working on was a small 5x3 (Or, well, 3x5. Whichever it is with 1 rod, and 8 blocks for cooling), using platinium blocks on the rod and gold as filler, and 2 access ports.

The reactor built fine the first time, but I placed the power tap in the wrong section so I broke it and moved it around. Reactor reformed, but clicking the controller returned a unknown error. Did this a couple times, decided to core the reactor and replace stuff figuring that one of the internal cooling blocks wasn't being recognized after swapping the controller and power tap around. Replaced the stuff, reactor formed up again, and then hard crash when trying to interact with the controller.

fwiw, this happened inside a compact machine.

**Comment by hexxod** [comment]
Got a link to them, was a server tick loop that crashed it.

https://pastebin.com/0Ex7ZFJz

Logging back in, I saw the reactor was still formed up but the power tap had for some reason turned into another controller.

**Comment by RogueLogix** [comment]
That's a different bug, and a new one too. However, I have no idea how to reproduce that to fix it.

**Comment by patthomasrick** [comment]
I experienced a very similar bug in singleplayer to hexxod's. I was building a reactor in a Compact Machine in ATM 7 0.4.2, placed the last block, and crashed immediately after. Log is below (I think I captured the important stuff).

<https://pastebin.com/8282WjZR>

When I rejoined, the reactor was completed/formed exactly as I placed it and everything seemed fine.

Also, disclaimer, I was using Optifine, so keep that in mind.

I tried reproducing the issue with a new Forge installation (40.1.0), Compact Machines 4.3.0, and Bigger Reactors 0.6.0 beta 1.1 (+Phosphophyllite 0.6.0 beta 1.2, Quartz 0.0.0 beta 1.3) and couldn't get it to crash/was able to create reactors in compact machines so I don't know how helpful this is...
